lib/crypto/base36.js

```diff
@@ -4,25 +4,31 @@
 
 'use strict'
 
-const crypto = require('crypto')
+const random = require('./random')
 
 function base36(len) {
   let out = []
-  while (out.length < len) {
-    let rand = crypto.randomBytes(len)
-    let randLen = rand.length
-    for (let i = 0; i < randLen; i++) {
-      let b = rand[i]
-      // 252-256 skews the base36 distribution, so skip those bytes
-      if (b < 252) {
-        out.push((b % 36).toString(36))
-        if (out.length === len) {
-          break
+  return random(len)
+    .then(bytes => {
+      let bytesLen = bytes.length
+      for (let i = 0; i < bytesLen; i++) {
+        let b = bytes[i]
+        // 252-256 skews the base36 distribution, so skip those bytes
+        if (b < 252) {
+          out.push((b % 36).toString(36))
         }
       }
-    }
-  }
-  return out.join('').toUpperCase()
+
+      if (len > bytesLen) {
+        return base36(len - bytesLen)
+      }
```

Comment:
In my tests, this sometimes produces fewer than the requested number of characters, e.g. I've observed base36(20) sometimes returning a string only 19 characters long.  ISTM this conditional should be checking `len` against the number of characters in `out`, rather than against the number of bytes originally generated?